rcl/src/rcl/wait.c

```diff
@@ -79,6 +80,9 @@ __wait_set_clean_up(rcl_wait_set_t * wait_set, rcl_allocator_t allocator)
     assert(ret == RCL_RET_OK);  // Defensive, shouldn't fail with size 0.
   }
   if (wait_set->impl) {
+    if (wait_set->impl->rmw_waitset) {
+      assert(rmw_destroy_waitset(wait_set->impl->rmw_waitset) == RMW_RET_OK);
```

Comment:
Another thing I'll mention here is that the other functions are implemented in this file and I can ensure they shouldn't fail in that scenario fairly easily. But the same might not be so easy to say of the `rmw_destroy_waitset` function since it is presumably implemented by each middleware implementation and might fail in some scenario. In which an assert might not be the right action to take here. But I think it's ok to handle that when it comes up and leave the assert for now.